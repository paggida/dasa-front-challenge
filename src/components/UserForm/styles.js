import styled from "styled-components";
import color from "../../styles/colors";
import metrics from "../../styles/metrics";

export const Form = styled.form`
  margin-top: ${metrics.baseMargin * 2 + metrics.base};
  width: 100%;
  max-width: 500px;
  display: flex;
  input {
    flex: 1;
    height: 55px;
    padding: 0 ${metrics.basePadding + metrics.base};
    background: ${color.white};
    font-size: ${metrics.fontSize * 2 + metrics.base};
    color: ${color.dark};
    border-radius: ${metrics.baseRadius + metrics.base};
    border: ${({ withError }) => (withError ? "2px solid " + color.error : 0)};
  }
  button {
    width: 155px;
    height: 55px;
    padding: 0 ${metrics.basePadding + metrics.base};
    margin-left: ${metrics.baseMargin * 2 + metrics.base};
    background: ${color.tertiary};
    color: ${color.white};
    border: 0;
    font-size: ${metrics.fontSize * 2 + metrics.base};
    font-weight: bold;
    border-radius: ${metrics.baseRadius + metrics.base};

    &:hover {
      background: ${color.quaternary};
      cursor: pointer;
    }
  }
`;
