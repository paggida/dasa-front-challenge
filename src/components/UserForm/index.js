import React, { Component } from "react";
import propTypes from "prop-types";
import { Form } from "./styles";
import api from "../../services/api";

export default class UserForm extends Component {
  state = {
    loading: false,
    invalidUser: false,
    user: ""
  };

  static propTypes = {
    collectionResponse: propTypes.array.isRequired
  };

  handleSearchRepositories = async e => {
    e.preventDefault();
    const { user } = this.state;
    //const { collectionResponse } = this.props;

    this.setState({ loading: true });

    this.searchRepositories(user)
      .then(data => {})
      .catch(() => {
        this.setState({ invalidUser: true });
      })
      .finally(() => {
        this.setState({ loading: false });
      });
  };

  searchRepositories = user =>
    new Promise(async (resolve, reject) => {
      api
        .get(`/users/${user}/repos`)
        .then(({ data }) => {
          resolve(data);
        })
        .catch(err => {
          reject(err);
        });
    });

  render() {
    const { loading, invalidUser, user } = this.state;
    return (
      <Form withError={invalidUser} onSubmit={this.handleSearchRepositories}>
        <input
          type="text"
          placeholder="Usuário do GitHub"
          value={user}
          onChange={e => this.setState({ user: e.target.value })}
        />
        <button type="submit">
          {loading ? (
            <i className="fa fa-spinner fa-pulse" />
          ) : (
            "Buscar repositórios"
          )}
        </button>
      </Form>
    );
  }
}
