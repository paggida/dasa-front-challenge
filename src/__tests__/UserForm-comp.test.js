import React from 'react';
import UserForm from '../components/UserForm';
import { mount } from 'enzyme';
import toJson from "enzyme-to-json"

it("Should render the user form", () => {
  const props = {
    handleOutput: ()=>{}
  };
  const comp = mount(<UserForm {...props} />);

  expect(toJson(comp)).toMatchSnapshot();
  comp.unmount();
});
