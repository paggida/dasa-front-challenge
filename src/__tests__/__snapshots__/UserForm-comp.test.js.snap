// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the user form 1`] = `
<UserForm
  handleOutput={[Function]}
>
  <styled.form
    onSubmit={[Function]}
    withError={false}
  >
    <form
      className="sc-AxjAm fciyZm"
      onSubmit={[Function]}
    >
      <input
        onChange={[Function]}
        placeholder="Usuário do GitHub"
        type="text"
        value=""
      />
      <button
        id="searchBtn"
        type="submit"
      >
        Buscar repositórios
      </button>
    </form>
  </styled.form>
</UserForm>
`;
